runtime/js/10_permissions.js

```diff
@@ -139,18 +138,15 @@ function cache(desc, state) {
     key += "$";
   }
   if (MapPrototypeHas(statusCache, key)) {
-    const status = MapPrototypeGet(statusCache, key);
-    if (status.state !== state) {
-      status.state = state;
-      status.status.dispatchEvent(new Event("change", { cancelable: false }));
+    const cachedStatus = MapPrototypeGet(statusCache, key);
+    if (cachedStatus.state === status.state) {
+      return cachedStatus;
     }
-    return status.status;
```

Comment:
This breaks `onchange`. We need an early return at this selection level or the cached value will be displaced by the code below. I have a test case (which we should probably add):
```ts
const status = await Deno.permissions.query({ name: "read", path: "foo" });
console.log(status.state);
status.onchange = () => console.log(status.state);
await Deno.permissions.request({ name: "read", path: "foo" }); // y
await Deno.permissions.revoke({ name: "read", path: "foo" });

// Expected output:
//
// prompt
// ✅ Granted read access to "foo".
// granted
// prompt
```